bake/hclparser/hclparser.go

```diff
@@ -514,3 +512,23 @@ func setLabel(v reflect.Value, lbl string) int {
 	}
 	return -1
 }
+
+func filterAttributesDiags(diags hcl.Diagnostics, reserved map[string]struct{}) hcl.Diagnostics {
+	var fdiags hcl.Diagnostics
+	for _, d := range diags {
+		if fout := func(d *hcl.Diagnostic) bool {
+			if d.Detail == "Blocks are not allowed here." {
+				return true
+			}
+			for r := range reserved {
+				if strings.HasPrefix(d.Detail, fmt.Sprintf(`Argument "%s" was already set at `, r)) {
```

Comment:
Add a comment for this about where this is coming from and why it can't be detected.